apps/mobile/src/utils/firebase.ts

```diff
@@ -27,16 +27,16 @@ export const fireCache: FireCache = {
 	idToken: undefined,
 };
 
-auth().onIdTokenChanged(async (user) => {
-	if (user) {
-		fireCache.idToken = await user.getIdToken();
-	} else {
-		fireCache.idToken = undefined;
-	}
-});
-
 export const initializeAuth = async () => {
 	if (auth().currentUser) {
 		fireCache.idToken = await auth().currentUser?.getIdToken();
+
+		auth().onIdTokenChanged(async (user) => {
+			if (user) {
+				fireCache.idToken = await user.getIdToken();
+			} else {
+				fireCache.idToken = undefined;
+			}
+		});
```

Comment:
I think putting event listener in a function scope might be not good. Prefer to put it at global/module scope

Reply:
hmm, agree -- you reminds me that we'll miss the case that use login after open the App, above code block won't covered that case, while always subscribe the change won't take that much effort/resource at global/module scope anyway.